internal/clients/team7/frameworks/OpinionFramework.go

```diff
@@ -1,31 +1,61 @@
 package frameworks
 
-/*
-Base interface for all opinion frameworks. This can be used to define
-different behaviours of opinion formation giving the agent different behaviours.
-*/
-type IOpinionFramework[I, SN, O any] interface {
-	GetOpinion(inputs I, socialNetwork SN) O
-}
+import (
+	"math"
 
-type OpinionFrameworkInputs struct {
-}
+	"github.com/google/uuid"
+)
 
-type OpinionFrameworkOutputs struct {
+type OpinionFrameworkInputs struct {
+	AgentOpinion map[uuid.UUID]float64
+	Mindset      float64
 }
 
 type OpinionFramework struct {
-	IOpinionFramework[OpinionFrameworkInputs, SocialNetwork, OpinionFrameworkOutputs]
-	inputs *OpinionFrameworkInputs
+	Inputs *OpinionFrameworkInputs
 }
 
-// Constructor for OpinionFramework
 func NewOpinionFramework(of OpinionFrameworkInputs) *OpinionFramework {
-	return &OpinionFramework{inputs: &of}
+	return &OpinionFramework{Inputs: &of}
 }
 
-func (of *OpinionFramework) GetOpinion(inputs OpinionFrameworkInputs, socialNetwork SocialNetwork) OpinionFrameworkOutputs {
-	of.inputs = &inputs
-	// TODO: Formulate opinion
-	return OpinionFrameworkOutputs{}
+func (of *OpinionFramework) GetOpinion() float64 {
```

Comment:
This should take the input rather than when creating the new Opinion Framework. We will use it multiple times so we should pass the input every time.